src/ClassDiscovery.php

```diff
@@ -2,126 +2,117 @@
 
 namespace Http\Discovery;
 
-use Puli\Discovery\Api\Discovery;
+use Http\Discovery\Exception\DiscoveryFailedException;
+use Http\Discovery\Exception\StrategyUnavailableException;
 
 /**
  * Registry that based find results on class existence.
  *
  * @author David de Boer <david@ddeboer.nl>
  * @author Márk Sági-Kazár <mark.sagikazar@gmail.com>
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
  */
 abstract class ClassDiscovery
 {
     /**
-     * @var GeneratedPuliFactory
+     * A list of strategies to find classes.
+     *
+     * @var array
      */
-    private static $puliFactory;
+    public static $strategies = [
+        Strategy\Puli::class,
+        Strategy\HttpClients::class,
```

Comment:
No, we do not **need** to have those as separate strategies. We could merge them, I had them separate because it looked good. (Highly opinionated in other words). I'll merge them because of https://github.com/php-http/discovery/pull/59#discussion_r68453992